addons/core/addon/components/filter-tags/index.js

```diff
@@ -26,24 +27,13 @@ export default class FilterTagsIndexComponent extends Component {
 
   // =methods
 
-  getCurrentQueryParams(key) {
-    let queryParamValue;
-    try {
-      queryParamValue = JSON.parse(this.router.currentRoute.queryParams[key]);
```

Comment:
Wow I think I know what the issue is and is just an ember quirk. Setting a "default" or initial query parameter value does what you would expect for the route since it just gets the value directly but I think ember considers that "default" value to be the state as if you had no query parameter set and if you could see the URL you would see in fact there is no query parameters for that default state.

e.g. `page: 1` you would get no query params but if you goto another page you would get `?page=2` and then if you go _back_ to page 1, the query param would disappear since ember considers that the "default" state which is no query params.

For sessions it probably looks like `/sessions` with no query params for the 3 default you have selected but if you change it to just active for example, it would probably be `?status=active` but if then if you change it back to have all 3, the query param disappears as that's the "default" state. 🤯

Reply:
thank you for the detailed explanation!